docs/design/hub-and-spoke/switch-validate/switch-validate.md

```diff
@@ -10,4 +10,284 @@ Here is a FSM diagram that helps to understand main steps of validation switch r
+    ],
+    "proper": [
+      <meterInfo>
+    ]
+  },
+  "rules": {
+    "excess": [-9223372036854776000],
+    "excess-hex": ["800000000000001b"],
+    "misconfigured": [-9223372036854776000],
+    "misconfigured-hex": ["800000000000001b"],
+    "missing": [-9223372036854776000],
+    "missing-hex": ["800000000000001b"],
+    "proper": [-9223372036854776000],
+    "proper-hex": ["800000000000001b"]
+  }
+}
+```
+
+## v2 API : GET
+
+### Validate the rules and the meters installed on the switch
+
+`GET https://{host}/v2/switches/{switch_id}/validate`
+
+Response payload
+
+```json
+{
+  "groups": {
+    "excess": [
+      <groupInfo>
+    ],
+    "misconfigured": [
+      <groupInfo>
+    ],
+    "missing": [
+      <groupInfo>
+    ],
+    "proper": [
+      <groupInfo>
+    ]
+  },
+  "logical_ports": {
+    "error": "some_error",
+    "excess": [
+      <logicalPortInfo>
+    ],
+    "misconfigured": [
+      <logicalPortInfo>
+    ],
+    "missing": [
+      <logicalPortInfo>
+    ],
+    "proper": [
+      <logicalPortInfo>
+    ]
+  },
+  "meters": {
+    "excess": [
+      <meterInfo>
+    ],
+    "misconfigured": [
+      <meterInfo>
+    ],
+    "missing": [
+      <meterInfo>
+    ],
+    "proper": [
+      <meterInfo>
+    ]
+  },
+  "rules": {
+    "excess": [
+        <ruleInfo>
+    ],
+    "misconfigured": [
+      {
+        "flow_id": "9",
+        "flow_path": "10a1df3a-dc30-453e-8671-246f82ea0e77",
+        <ruleInfo>
+      }
+    ],
+    "missing": [
+      {
+        "flow_id": "10",
+        "flow_path": "e0b0716e-cc59-4ef6-8572-518bdcdc5c72",
+        <ruleInfo>
+      }
+    ],
+    "proper": [
+      {
+        "flow_id": "11",
+        "flow_path": "1cdc6db3-76b6-4775-935d-c317f103df44",
+        <ruleInfo>
+      }
+    ]
+  }
+}
+```
+
+#### groupInfo
+
+```json
+{
+  "excess_group_buckets": [
+    {
+      "port": 2,
+      "vlan": 2,
+      "vni": 2
+    }
+  ],
+  "group_buckets": [
+    {
+      "port": 3,
+      "vlan": 3,
+      "vni": 3
+    }
+  ],
+  "group_id": 10,
+  "missing_group_buckets": [
+    {
+      "port": 4,
+      "vlan": 4,
+      "vni": 4
+    }
+  ]
+}
+```
+
+#### logicalPortInfo
+
+```json
+{
+  "actual": {
+    "physical_ports": [10, 11]
+  },
+  "expected": {
+    "physical_ports": [10, 11, 13]
+  },
+  "logical_port_number": 100,
+  "type": "lag"
+}
+```
+
+#### meterInfo
+
+```json
+{
+  "actual": {
+    "flags": ["KBPS"],
+    "rate": 250
+  },
+  "expected": {
+    "flags": ["STATS"],
+    "rate": 200
+  },
+  "burst_size": 200,
+  "cookie": 10,
+  "flow_id": "10",
+  "meter_id": 1
+}
+```
+
+#### ruleInfo
+
+```json
+{
+  "ruleId": "0x8000000000000002",
+  "cookie": -9223372036854776000,
+  "cookie-hex": "800000000000001b",
+  "flags": [
+    "RESET_COUNTERS"
+  ],
+  "match": [
+    {
+      "value": 1,
+      "mask": 1,
+      "field": "IN_PORT"
+    }
+  ],
+  "instructions": {
+    "applyActions": [
+      "SET_FIELD"
+    ],
+    "writeActions": [
+      "SET_FIELD"
+    ]
```

Comment:
For the whole picture it's better to show whole possible fields in `expected` and `actual` fields